### PR TITLE
Function DeleteConfigMap and DeleteSecret are inlined

### LINKED ISSUE #11427: Inline functions from test/utils.go
**Is your feature request related to a problem? Please describe**:

The functions `DeleteConfigMap` and `DeleteSecret` in `tests/utils.go` could be inlined, since the do not provide additional value. Also since this is test code, assertions should be used instead of panicking on errors.

**Describe the solution you'd like**:

Inline `DeleteConfigMap` and `DeleteSecret` into the places where they are used.

### PR DESCRIPTION
**Pull request by anishbista60**
### Description
Function `DeleteConfigMap` and ` DeleteSecret` present in  `tests/utils.go` are inlined into the place where they are used.
### What this PR does
Before this PR:

The functions `DeleteConfigMap` and `DeleteSecret` in `tests/utils.go` doesnot look so useful because these function being called again and again in `tests/config_test.go`.

After this PR:

 Then, These function are removed from `tests/utils.go` and inlined into the `tests/config_test.go` where they are used.

Fixes #11427

### Why we need it and why it was done in this way
- Because it removes the overhead of the function call itself. 
- The second is it permits the compiler to more effectively apply other optimisation strategies.

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/config_test.go
Comment (alicefr) [inline tests/config_test.go:159]: Please, inline the error in the if statement

File: tests/config_test.go
Comment (0xFelix) [inline tests/config_test.go:99]: Can you run formatting over this please?

File: tests/config_test.go
Comment (0xFelix) [inline tests/config_test.go:518]: Please drop this change

**Review by 0xFelix — COMMENTED** [review]
Can you squash your commits please?

**Comment by 0xFelix** [comment]
Please run `make format` and make sure your commit includes a meaningful title and message.

**Comment by anishbista60** [comment]
@0xFelix done sir

**Comment by 0xFelix** [comment]
Looks good, but needs a rebase. Also, please put something into the body of the commit message. Thank you!

**Comment by anishbista60** [comment]
@0xFelix Except squash commit . Is here any extra changes to be done ?

**Comment by anishbista60** [comment]
@0xFelix Everything looks fine . good to go for futher test

**Comment by 0xFelix** [comment]
@anishbista60 there is still no message in your commit, only a title?

**Comment by anishbista60** [comment]
@0xFelix commit message and title mean ? not getting sir . -m flag is used for writing message right and what about tilte? could you please give me hint ?

**Review by 0xFelix — APPROVED** [review]
/lgtm

Thank you very much! Now let's find an approver.

@alicefr would you mind giving this a look?

**Comment by anishbista60** [comment]
@0xFelix Done

**Review by 0xFelix — APPROVED** [review]
/lgtm

Thanks!

**Comment by 0xFelix** [comment]
Ping @alicefr again

**Comment by alicefr** [comment]
@anishbista60 your git tree isn't clean, please, commit all your changes. You can see the result in the CI job `
pull-kubevirt-build `

**Comment by anishbista60** [comment]
@0xFelix @alicefr can you please look here . if it is ok then i will squashed  the commit or else will do what things is required

**Comment by alicefr** [comment]
@anishbista60 why did you need to modify the bazel files?

File: tests/config_test.go
Comment (alicefr) [inline tests/config_test.go:165]: nit: additional empty line

File: tests/config_test.go
Comment (alicefr) [inline tests/config_test.go:263]: nit: additional empty line

File: tests/config_test.go
Comment (victortoso) [inline tests/config_test.go:101]: Would you mind 

```diff
-                               if err := kubevirt.Client().CoreV1().ConfigMaps(testsuite.GetTestNamespace(nil)).Delete(context.Background(), configMapName, metav1.DeleteOptions{}); err != nil {
+                               configMap := kubevirt.Client().CoreV1().ConfigMaps(testsuite.GetTestNamespace(nil))
+                               if err := configMap.Delete(context.Background(), configMapName, metav1.DeleteOptions{}); err != nil {
```

Reduces the line length from 178 to 117. We don't have a defined limit but anything above 150 is a lot imho.

This comments applies to other changes, the problem is that `configMap` is already used in other places. 
Hm. The return type of `ConfigMaps()`  is an interface, so perhaps `configMapIface`? Not sure, I'm terrible with names :)

**Review by victortoso — COMMENTED** [review]
Please do run `make generate` and include the changes in your commit.

One last nit, the body log of your commit message is a long line too, feel free to break it in multiple lines.

**Comment by anishbista60** [comment]
@victortoso  please check it

**Comment by victortoso** [comment]
The problem here is that you need to run `make generate` and add the changes to your commit. Running `make generate` outputs:

```diff
toso@tapioca ~/s/k/kubevirt (review-11450)> git diff
diff --git a/tests/config_test.go b/tests/config_test.go
index 1577b00d78..89d3cfef43 100644
--- a/tests/config_test.go
+++ b/tests/config_test.go
@@ -36,18 +36,18 @@ import (
        "golang.org/x/crypto/ssh"

        "k8s.io/apimachinery/pkg/api/errors"
-       v1 "kubevirt.io/api/core/v1"
        metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
+       v1 "kubevirt.io/api/core/v1"
+
        "kubevirt.io/kubevirt/pkg/config"
        "kubevirt.io/kubevirt/tests"
        "kubevirt.io/kubevirt/tests/console"
        "kubevirt.io/kubevirt/tests/decorators"
        "kubevirt.io/kubevirt/tests/exec"
+       "kubevirt.io/kubevirt/tests/framework/kubevirt"
        "kubevirt.io/kubevirt/tests/libpod"
        "kubevirt.io/kubevirt/tests/libvmi"
        "kubevirt.io/kubevirt/tests/testsuite"
-       "kubevirt.io/kubevirt/tests/framework/kubevirt"
 )

 var _ = Describe("[rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-compute]Config", decorators.SigCompute, func() {
diff --git a/tests/utils.go b/tests/utils.go
index dfd3c33948..867daee27e 100644
--- a/tests/utils.go
+++ b/tests/utils.go
@@ -147,7 +147,6 @@ func CreateSecret(name, namespace string, data map[string]string) {
        }
 }

-
 func RunVMIAndExpectLaunch(vmi *v1.VirtualMachineInstance, timeout int) *v1.VirtualMachineInstance {
        vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi)
        Expect(err).ToNot(HaveOccurred())
```

**Comment by anishbista60** [comment]
@victortoso  Haven't ran `make generate`
Getting permission denied
`hack/dockerized hack/build-ginkgo.sh
/usr/bin/env: ‘bash\r’: Permission denied
make: *** [Makefile:48: generate] Error 126`

**Comment by victortoso** [comment]
> API rule violation: names_match,kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1,ImportProxy,HTTPProxy

@anishbista60 no clue how you got there from the current `inline-01` branch. If you update the branch I can try to help you navigate what's going on.

**Comment by anishbista60** [comment]
@victortoso yeah done please checkout it is due my problem in my laptop and now ,iam using my friends mac

**Comment by anishbista60** [comment]
@victortoso can you please test other jobs?

File: tests/config_test.go
Comment (victortoso) [reply tests/config_test.go:38]: @anishbista60  you are missing this request
```diff
--- a/tests/config_test.go
+++ b/tests/config_test.go
@@ -34,8 +34,8 @@ import (
        . "github.com/onsi/ginkgo/v2"
        . "github.com/onsi/gomega"
        "golang.org/x/crypto/ssh"
-
        "k8s.io/apimachinery/pkg/api/errors"
+
        metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
        v1 "kubevirt.io/api/core/v1"
```

File: tests/config_test.go
Comment (victortoso) [reply tests/config_test.go:101]: You ignored this comment but it is fine, we don't have any hard requirements for line length anyway.

**Review by victortoso — COMMENTED** [review]
Please address the request from Felix so we can move forward with this.

**Comment by anishbista60** [comment]
@victortoso  please check it

**Comment by victortoso** [comment]
/lgtm cancel

Could you please break the line of the commit log?

```diff
-The function DeleteConfigMap and DeleteSecret does not provide much additional value. Additionally, assertions should be used instead of panicking on errors.
+The function DeleteConfigMap and DeleteSecret does not provide much additional value. 
+Additionally, assertions should be used instead of panicking on errors.
```

**Comment by anishbista60** [comment]
@victortoso  done sir

**Comment by anishbista60** [comment]
@victortoso could you please go for futher testing?

**Comment by 0xFelix** [comment]
It needs approval to run the remaining tests. @alicefr Can you please take a look?

**Comment by alicefr** [comment]
@anishbista60 you are missing [this comment](https://github.com/kubevirt/kubevirt/pull/11450#discussion_r1521399588)  from @0xFelix

**Comment by alicefr** [comment]
The changes looks fine
/approve

**Comment by anishbista60** [comment]
@0xFelix  what's next

**Comment by victortoso** [comment]
@anishbista60 a little bit of patience, please. This is already on merge pool, it'll be merged. Thank you.

**Comment by anishbista60** [comment]
@0xFelix Thanks for helping me in my first PR. Want to become contributor to kubevirt.

**Pull request merged**
